firmware/Core/Src/telecommands/log_telecommand_defs.c

```diff
@@ -195,3 +197,76 @@ uint8_t TCMDEXEC_log_set_system_debugging_messages_state(const char *args_str, T
+    const char *log_text = LOG_get_most_recent_log_message_text();
+    switch (tcmd_channel) {
+        case TCMD_TelecommandChannel_DEBUG_UART:
+            LOG_to_umbilical_uart(log_text);
+            break;
+        case TCMD_TelecommandChannel_RADIO1:
+            LOG_to_uhf_radio(log_text);
+            break;
+        default:
+            // Channel is unknown, log just to the filesystem
+            LOG_message(LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_ERROR, LOG_SINK_FILE, "TCMDEXEC_log_report_latest_message(): Unknown telecommand channel");
+            break;
+    }
+
+    return 0;
+}
+
+/// @brief Telecommand: Report the N latest log messages to the incoming
+/// telecommand channel
+/// @param args_str
+/// - Arg 0: Number of latest log messages to report
+uint8_t TCMDEXEC_log_report_n_latest_messages_from_memory(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                        char *response_output_buf, uint16_t response_output_buf_len) {
+    // This does not get logged. Send result directly back on telecommand
+    // channel the request arrived on
+    uint64_t requested_number_of_entries = 0;
+    uint8_t result = TCMD_extract_uint64_arg(args_str, strlen(args_str), 0, &requested_number_of_entries);
+    if (result) {
+        LOG_message(LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_ERROR, LOG_SINK_ALL, "TCMDEXEC_log_report_n_latest_messages(): Unable to parse number of entries from first telecommand argument");
+        return 1;
+    }
+    const uint8_t max_entries = LOG_memory_table_max_entries();
+    if (requested_number_of_entries > max_entries) {
+        LOG_message(LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_ERROR, LOG_SINK_ALL, "TCMDEXEC_log_report_n_latest_messages(): Requested number of log messages must be at most %d", max_entries);
+        return 1;
+    }
+
+    // Report the entries in the log table, which is a circular buffer
+    uint8_t current_index = LOG_get_memory_table_index_of_most_recent_log_entry();
+    int16_t start_index = ((int16_t)current_index + 1 - (int16_t)requested_number_of_entries); 
+    if (start_index < 0) {
+        start_index += max_entries;
+    }
+    void (*transmit_function)(const char *);
+    switch (tcmd_channel) {
+        case TCMD_TelecommandChannel_DEBUG_UART:
+            transmit_function = LOG_to_umbilical_uart;
+            break;
+        case TCMD_TelecommandChannel_RADIO1:
+            transmit_function = LOG_to_uhf_radio;
+            break;
+        default:
+            // Channel is unknown, log error to the filesystem and return
+            LOG_message(LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_ERROR, LOG_SINK_FILE, "TCMDEXEC_log_report_latest_message(): Unknown telecommand channel");
+            return 1;
+            break;
+    }
+
+    // Report the messages
+    for (uint8_t response_number = 0; response_number < requested_number_of_entries; response_number++) {
+        const uint16_t log_index = (start_index + response_number) % max_entries;
```

Comment:
It might be fine, but I have a strange fear that the `start_index + response_number` intermediate value may be cast to a `uint8` (as response_number is a uint8). I can't find a source that confirms this one way or another. For safety (unless we can confirm it's fine), would you mind making `response_number` a uint16 or int16?

I've approved the PR, and you can merge it if you're highly confident this intermediate value can't/won't overflow or wrap around.

Reply:
That was an oversight. Would rather use 16-bit numbers to be safe. Updated, and ready to approve.